feat(ci): Add support for managing Halyard's versions.yml (#195)
spinnaker/buildtool: README.md

    1. Raise a PR to:
       [github.com/spinnaker/spinnaker.io](https://github.com/spinnaker/spinnaker/io/pulls)

### Fetch versions.yml

Halyard uses `versions.yml` as the source of truth for available Spinnaker

> versions to deploy. Fetch the file from GCS bucket for editing.

Reviewer: ```suggestion
versions to deploy. Fetch the file from GCS for editing.
```
or maybe "its GCS bucket" ?

```
./dev/buildtool.sh fetch_versions
```

To troubleshoot, check the fetched file.

```
$ cat output/fetch_versions/versions.yml
illegalVersions:
- reason: Broken apache config makes the UI unreachable
  version: 1.2.0
- reason: UI does not load
  version: 1.4.0
latestHalyard: 1.49.0
latestSpinnaker: 1.28.1
versions:
- alias: v1.28.1
  changelog: https://spinnaker.io/changelogs/1.28.1-changelog/
  lastUpdate: 1661279166000
  minimumHalyardVersion: '1.45'
  version: 1.28.1
  <snip>
```

### Update versions.yml

> Add the new Spinnaker release version to `versions.yml`.

Reviewer: ```suggestion
Add a new Spinnaker version to `versions.yml`.
```

```
version=1.27.0
halyard_version=1.45.0
latest_halyard_version=1.51.0 # optional
versions_yml_path=source_code/update_versions/versions.yml

./dev/buildtool.sh update_versions \
  --versions_yml_path "${versions_path}" \
  --spinnaker_version "${version}" \
  --minimum_halyard_version "${halyard_version}" \
  --latest_halyard_version "${latest_halyard_version}"
```

To troubleshoot, check input and output `versions.yml` files.

```
diff \
  source_code/fetch_versions/versions.yml \
  source_code/update_versions/versions.yml \
```

### Publish versions.yml

> When ready to release the new Spinnaker version to users, publish the

Reviewer: ```suggestion
When ready to release a new Spinnaker version to users, publish the
```

> `versions.yml` file to GCS bucket.

Reviewer: ```suggestion
`versions.yml` file to GCS.
```
